File: Doeunnkimm/8주차/p_N으로표현.js
function solution(N, number) {
  const use = Array.from(Array(9), () => new Set())

  if (N === number) return 1

  use.forEach((el, idx) => {
    if (idx !== 0) el.add(Number(String(N).repeat(idx)))
  })

  for (let i = 1; i <= 8; ++i) {
    for (let j = 1; j < i; ++j) {
      for (let first of use[j]) {
        for (let second of use[i - j]) {
          use[i].add(first + second)
          use[i].add(first - second)
          use[i].add(first * second)
          use[i].add(first / second)
        }
      }
    }
    if (use[i].has(number)) return i
  }
  return -1
}

File: seungri2/8주차/p_n으로 표현.py
def solution(N, number):
    dp = [set() for i in range(9)]

    for i in range(1, 9):
        dp[i].add(int(str(N)*i))

        for j in range(i//2 + 1):
            for first in dp[j]:
                for second in dp[i-j]:
                    dp[i].add(first + second) # 덧셈 연산
                    dp[i].add(first - second) # 뺄셈 연산
                    dp[i].add(second - first) # 뺄셈 연산
                    dp[i].add(first * second) # 곱셈 연산
                    if second != 0 :
                        dp[i].add(first // second) # 나눗셈 연산
                    if first != 0 :
                        dp[i].add(second // first) # 나눗셈 연산
                    
        if number in dp[i]:
            return i
    return -1